lightning/src/routing/scoring.rs

```diff
@@ -225,6 +225,16 @@ impl<'a, T: Score + 'a> LockableScore<'a> for MultiThreadedLockableScore<T> {
 	}
 }
 
+#[cfg(c_bindings)]
+impl<T: Score> Writeable for MultiThreadedLockableScore<T> {
+	fn write<W: Writer>(&self, writer: &mut W) -> Result<(), io::Error> {
+		self.lock().write(writer)
+	}
+}
+
+#[cfg(c_bindings)]
+impl<'a, T: Score + 'a> WriteableScore<'a> for MultiThreadedLockableScore<T> {}
```

Comment:
I'm a bit confused: isn't this redundant to this more generic implementation here?:
```
#[cfg(not(c_bindings))]
impl<'a, T> WriteableScore<'a> for T where T: LockableScore<'a> + Writeable {}
```
Or to ask differently: couldn't we just remove that `#[cfg(not(c_bindings))]` and have a implementation to the same effect?